### DISCUSSION TITLE
提供了这么多策略，每次只能选一个使用？我全都要，怎么组合？

### DISCUSSION THREAD

**Discussion by ShineKidd** [alovajs/alova#595] [Feature Request]
### What problem does the new feature need to solve?

如题

### How should the new features you expect looks like?

useRetriableRequest + useAutoRequest + 轮询这不是很常见的组合吗？

### How important is it to you?

Better to have

**Comment by MeetinaXD** [comment]
`useAutoRequest` 实现的效果就是轮询，一般是定时拉取数据刷新状态。

其次，`useRetriableRequest` 是自动失败重试，请求成功则不重试。一般用于提交操作。

有什么场景会需要组合二者使用吗？

**Reply by ShineKidd** [reply]
useAutoRequest 确实支持轮询，这个没注意看。
由于网络波动或者服务波动，请求失败后，是有必要自动重试的，不需要抛出异常，这个和 get post 请求方法无关。

**Reply by MeetinaXD** [reply]
不理解为什么轮询和请求重试要搭配使用，也想不到实际的应用情景。

退一步说，这个应用场景也是比较小众的，我们没办法很好地照顾到每一个需求。

不过，这个需求实现起来也比较简单，基于 `useRetriableRequest` 封装一个 hook 即可，这样也保留了最大的灵活性。
